File: gridlayoutjava/src/main/java/com/codekul/gridlayoutjava/MainActivity.java
package com.codekul.gridlayoutjava;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.GridLayout;
import android.widget.GridView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ArrayList<myItem> dataset = new ArrayList<>();
        dataset.add(new myItem(R.drawable.ic_launcher_background,"background0"));
        dataset.add(new myItem(R.drawable.ic_launcher_background,"background1"));
        dataset.add(new myItem(R.drawable.ic_launcher_background,"background2"));
        dataset.add(new myItem(R.drawable.ic_launcher_background,"background3"));
        dataset.add(new myItem(R.drawable.ic_launcher_background,"background4"));
        dataset.add(new myItem(R.drawable.ic_launcher_background,"background5"));



        myAdapter adapter = new myAdapter(this,dataset);

        ((GridView)findViewById(R.id.grid)).setAdapter(adapter);
    }
}

File: services&media/src/main/java/com/codekul/servicesmedia/MainActivity.kt
package com.codekul.servicesmedia

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
      var i = Intent(this,MyService::class.java)
        btnplay.setOnClickListener {
            startService(i)
        }
        btnStop.setOnClickListener {
            stopService(i)

        }
    }
}
